### PR TITLE
add check to gateway regression test

### PR DESCRIPTION
**Pull request by mitchdraft**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1159

### CODE REVIEW

File: ci/test-cluster/namespace-deleter.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  schedule: "*/15 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          serviceAccountName: namespace-deleter
+          containers:
+            - name: kubectl
+              image: bitnami/kubectl:1.15
+              # delete test namespaces if older than 30 minutes
```
Comment (marcogschmidt) [inline ci/test-cluster/namespace-deleter.yaml]:
```suggestion
              # delete test namespaces if older than 60 minutes
```

File: test/kube2e/gateway/gateway_suite_test.go [hunk 2]
```diff
@@ -67,6 +71,16 @@ func StartTestHelper() {
...
+	Eventually(func() error {
+		ok, err := check.CheckResources(&options.Options{})
+		if err != nil {
+			return errors.Wrap(err, "unable to run glooctl check")
+		}
+		if ok {
+			return nil
+		}
+		return errors.New("glooctl check detected a problem with the installation")
+	}, "20s", "4s").Should(Not(BeNil()))
```
Comment (marcogschmidt) [inline test/kube2e/gateway/gateway_suite_test.go]: Shouldn't this be:

```suggestion
	}, "20s", "4s").Should(BeNil())
```

Comment (mitchdraft) [reply test/kube2e/gateway/gateway_suite_test.go]: 😮

**Pull request merged**